File: javascript/src/get-pis-auth-url-without-created-payee.js
import config from './config.js';
import { Moneyhub } from '@mft/moneyhub-api-client';
import yargs from 'yargs';
import { hideBin } from 'yargs/helpers';

const args = yargs(hideBin(process.argv)).option('state', {
  alias: 's',
  describe: 'state from original authorisation URL',
  default: 'foo',
}).argv;

const main = async () => {
  const moneyhub = await Moneyhub(config);
  const url = await moneyhub.getPaymentAuthorizeUrl({
    state: args.state,
    amount: 1000,
    bankId: 'test',
    payeeRef: 'reference',
    payerRef: 'reference',
    payee: {
      accountNumber: '12345678',
      name: 'payee-name',
      sortCode: '123456',
    },
  });
  console.log(url);
};

main();

File: python/src/get_pis_auth_url_without_created_payee.py
from config import *
import utils
import argparse

parser = argparse.ArgumentParser(description='Get PIS authorisation URL that creates a payee')
parser.add_argument("-s", "--state", default="foo", help="The state value")
args = parser.parse_args()

identity_server = config["identity_service_url"]

client_id = config["client"]["client_id"]
params = {
  "client_id": client_id,
  "scope": "openid id:test payment",
  "state": args.state,
  "redirect_uri": config["client"]["redirect_uri"],
  "response_type": "code",
  "claims": {
    "id_token": {
      "mh:con_id": {
        "essential": True
      },
      "mh:payment": {
        "essential": True,
        "value": {
          "amount": 100,
          "payeeRef": "reference",
          "payerRef": "reference",
          "payee": {
            "name": "payee name",
            "sortCode": "123456",
            "accountNumber": "12345678"
          }
        }
      }
    }
  },
  "aud": "{}/oidc".format(identity_server),
  "iss": client_id,
  **utils.get_common_claims()
}

request_uri = utils.post_request_object(config, params)
print("{}/oidc/auth?request_uri={}".format(identity_server, request_uri))